api/v1beta1/openstackmachine_types.go

```diff
@@ -33,6 +33,60 @@ const (
 	IPClaimMachineFinalizer = "openstackmachine.infrastructure.cluster.x-k8s.io/ip-claim"
 )
 
+// SchedulerHintValueType is the type that represents allowed values for the Type field.
+// +kubebuilder:validation:Enum=Bool;String;Number
+type SchedulerHintValueType string
+
+// Constants representing the allowed types for SchedulerHintAdditionalValue.
+const (
+	SchedulerHintTypeBool   SchedulerHintValueType = "Bool"
+	SchedulerHintTypeString SchedulerHintValueType = "String"
+	SchedulerHintTypeNumber SchedulerHintValueType = "Number"
+)
+
+// SchedulerHintAdditionalValue represents the value of a scheduler hint property.
+// The value can be of various types: Bool, String, or Number.
+// The Type field indicates the type of the value being used.
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'Bool' ? has(self.bool) : !has(self.bool)",message="bool is required when type is Bool, and forbidden otherwise"
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'Number' ? has(self.number) : !has(self.number)",message="number is required when type is Number, and forbidden otherwise"
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'String' ? has(self.string) : !has(self.string)",message="string is required when type is String, and forbidden otherwise"
+// +union.
+type SchedulerHintAdditionalValue struct {
+	// Type represents the type of the value (Bool, String, or Number).
+	// +kubebuilder:validation:Required
+	// +unionDiscriminator
+	Type SchedulerHintValueType `json:"type"`
+
+	// Bool is the boolean value of the scheduler hint, used when Type is "Bool".
+	// This field is required if type is 'Bool', and must not be set otherwise.
+	// +unionMember,optional
+	Bool *bool `json:"bool,omitempty"`
+
+	// Number is the integer value of the scheduler hint, used when Type is "Number".
+	// This field is required if type is 'Number', and must not be set otherwise.
+	// +unionMember,optional
+	Number *int `json:"number,omitempty"`
```

Comment:
Minimum or maximum values that are applicable here?

Reply:
I don't think there can be any minima or maxima in this case, as the nature of the interface is we don't know what it will be used for.

I wonder if we should make the size explicit anyway by making this an int64.

Reply:
You can set the maximum to the maximum of an int64 so that it is at least in the schema, but, ideally it would be lower.

All integer types in Kube APIs should be int32 or int64, other int primitives are not allowed by convention

Reply:
I can see that openapi codegen set `int32` as default format for `int` typed values in zz_generated.openapi.go schema.
As long as the schema set it as int32, do you still need to set max or min value using the max and min value of int32 explicitly?

Reply:
You should always use the typed int, so, int32 or int64 in this case. And its best practice to have maximums, but in this case, I don't know that it's possible without excluding actually valid OSP API values